app/src/main/java/com/example/tanya/godeltechchallengeandroid/api/ApiModule.kt

```diff
@@ -3,9 +3,15 @@ package com.example.tanya.godeltechchallengeandroid.api
 import android.content.Context
 import android.content.Context.MODE_PRIVATE
 import android.content.SharedPreferences
+import com.example.tanya.godeltechchallengeandroid.api.rest.RestService
 import dagger.Binds
 import dagger.Module
 import dagger.Provides
+import io.reactivex.schedulers.Schedulers
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
 
 @Module
 abstract class ApiModule {
```

Comment:
You could simplify the class implementation with:
1) move the bindings logic into separate interface which is marked with the `@Module` annotation and include it in the ApiModule
2) make the ApiModule class not-abstract, move all logic from companion object, remove the "feints" with jvm bytecode.